[iOS] Platform specifics for ListView Bounces

adrianknight89 opened the pull request:
### Description of Change

Added platform-specific option to turn on and off bouncing on iOS `ListView`. If bouncing is off, then pull-to-refresh will be disabled. This is by iOS.
### API Changes

Added:
- public static readonly BindableProperty BouncesProperty = BindableProperty.Create(nameof(Bounces), typeof(bool), typeof(ListView), true);
- public static bool GetBounces(BindableObject element)
- public static void SetBounces(BindableObject element, bool value)
- public static bool Bounces(this IPlatformElementConfiguration<iOS, FormsElement> config)
- public static IPlatformElementConfiguration<iOS, FormsElement> SetBounces(this IPlatformElementConfiguration<iOS, FormsElement> config, bool value)
- public static IPlatformElementConfiguration<iOS, FormsElement> EnableBounces(this IPlatformElementConfiguration<iOS, FormsElement> config)
- public static IPlatformElementConfiguration<iOS, FormsElement> DisableBounces(this IPlatformElementConfiguration<iOS, FormsElement> config)
### PR Checklist
- [ ] Has tests (if omitted, state reason in description)
- [x] Rebased on top of master at time of PR
- [x] Changes adhere to coding standard
- [x] Consolidate commits as makes sense

StephaneDelcroix commented on Xamarin.Forms.Core/PlatformConfiguration/iOSSpecific/ListView.cs:
```diff
@@ -0,0 +1,42 @@
+
+	public static class ListView
+	{
+		public static readonly BindableProperty BouncesProperty = BindableProperty.Create(nameof(Bounces), typeof(bool), typeof(ListView), true);
```
if it's boolean property, should be IsBouncing, or HasBounces, CanBounces, IsBouncesEnabled,....

jassmith replied:
IsBounceEnabled

adrianknight89 commented:
@jassmith @StephaneDelcroix  I changed to `IsBounceEnabled` and removed enable/disable methods. Is it required that platform-specific options have all 6 methods as shown in @samhouts's `IsNavigationBarTranslucent`? I think if we are dealing with a `bool`, setting the property directly should be preferable to the extra enable/disable methods.

rmarinho commented on Xamarin.Forms.Core/Xamarin.Forms.Core.csproj:
```diff
@@ -89,10 +89,9 @@
     <Compile Include="EnumerableExtensions.cs" />
     <Compile Include="PlatformConfiguration\AndroidSpecific\AppCompat\Application.cs" />
     <Compile Include="PlatformConfiguration\AndroidSpecific\Application.cs" />
-    <Compile Include="PlatformConfiguration\AndroidSpecific\TabbedPage.cs" />
```
Were these files removed from csproj why ?

adrianknight89 replied:
Sorry. I had troubles with rebase & merge recently that affected this and another PR. It should be fixed.

rmarinho commented:
needs rebase

rmarinho commented:
Also needs approval from @jassmith and @StephaneDelcroix

adrianknight89 commented:
Done.

jassmith commented:
This can easily be implemented in a third party library using an effect, in particular we believe this would be a great addition for this toolkit. https://github.com/FormsCommunityToolkit/Effects

Moving forward our policy with PlatformSpecifics is that all platform specifics which *can* be implemented without being in core should be. Only those which need to modify how core renderers work in a way that cannot be done with an effect will be accepted in core.
